shard-operator-integration-tests/src/test/java/com/redhat/service/bridge/shard/operator/cucumber/steps/BridgeIngressSteps.java

```diff
@@ -0,0 +1,73 @@
+package com.redhat.service.bridge.shard.operator.cucumber.steps;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.concurrent.TimeoutException;
+
+import com.redhat.service.bridge.shard.operator.cucumber.common.Context;
+import com.redhat.service.bridge.shard.operator.cucumber.common.TimeUtils;
+import com.redhat.service.bridge.shard.operator.resources.BridgeIngress;
+
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+/**
+ * Step definitions related to BridgeIngress
+ */
+public class BridgeIngressSteps {
+
+    private Context context;
+
+    public BridgeIngressSteps(Context context) {
+        this.context = context;
+    }
+
+    @When("^deploy BridgeIngress:$")
+    public void deployBridgeIngress(String bridgeIngressYaml) {
+        InputStream resourceStream = new ByteArrayInputStream(bridgeIngressYaml.getBytes(StandardCharsets.UTF_8));
```

Comment:
We must probably have unique `TemplateProvider` factories for those. This way we can easily change how we read these templates. We are looking for using Qute in the near future.

Reply:
Could be used to share same image for all BridgeIngress instances for tests using proper image.